### PR TITLE
Improve Make workflow

### PR DESCRIPTION
**Pull request by gssbzn**
Simplify some of the make workflow for linting and testing
Remove duplication golintci handles
Add a make step to set up lintci

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
-
-errcheck:
-	@sh -c "'$(CURDIR)/scripts/errcheck.sh'"
+.PHONY: test
 
 lint:
 	@echo "==> Linting all packages..."
 	golint -set_exit_status ./...
-	GOGC=30 golangci-lint run ./...
+	golangci-lint run ./... -E gofmt -E golint
```
Comment (gssbzn) [inline GNUmakefile:22]: `golangci` already handles errcheck (enable by default) and can handle gofmt and golint if told to so no need for separate commands

File: .drone.yml [hunk 2]
```diff
@@ -1,19 +1,35 @@
...
 type: docker
-name: linux-amd64
+name: pcgc
 
 platform:
   arch: amd64
   os: linux
 
 steps:
+  - name: test
```
Comment (gssbzn) [inline .drone.yml:11]: a bit of a split, should improve the speed a little, should also help to no longer install `golangci` as they provide a docker image we can use here `golangci`
I've been doing something similar for my project

File: .githooks/pre-commit [hunk 3]
```diff
@@ -1,3 +1,3 @@
 $!/usr/bin/env bash
 echo "==> Running pre-commit git hook..."
-make vet fmt fmtcheck errcheck lint
+make fmt test lint
```
Comment (gssbzn) [inline .githooks/pre-commit:3]: as commented before, `lint` already handles `vet`, `fmtcheck` and `errcheck`

File: GNUmakefile [hunk 4]
```diff
@@ -40,28 +32,27 @@ test-compile:
 	fi
 	@echo "==> Compiling test binary..."
 	go test -c $(TEST) $(TESTARGS)
-
-.PHONY: build test testacc vet fmt fmtcheck errcheck lint test-compile init-git-hooks
```
Comment (gssbzn) [inline GNUmakefile:44]: this was out of sync with available commands so I decided to define hem next to the command so we keep them updated

File: .drone.yml [hunk 5]
```diff
@@ -1,19 +1,35 @@
...
+    image: golang:1.13
+    volumes:
+      - name: deps
+        path: /go
+    commands:
+      - make test
+  - name: lint
+    image: golangci/golangci-lint:v1.21.0
+    commands:
+      - make lint
```
Comment (MihaiBojin) [inline .drone.yml:21]: does this not need the volume?

File: GNUmakefile [hunk 6]
```diff
@@ -40,28 +32,27 @@ test-compile:
...
+.PHONY: test-compile
 
 # Build targets
 clean:
 	@echo "==> Cleaning build artifacts..."
-	rm -rf out
 	go clean ./...
+.PHONY: clean
```
Comment (MihaiBojin) [inline GNUmakefile]: I think .PHONY should go above each target

File: GNUmakefile [hunk 7]
```diff
@@ -1,36 +1,28 @@
...
-GOFMT_FILES?=$$(find . -name '*.go' |grep -v vendor)
+export GO111MODULE := on
+TEST?=$$(go list ./...)
+GOFMT_FILES?=$$(find . -name '*.go')
 
 default: build
 
-test: fmtcheck
+setup:
+    curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s v1.21.0
```
Comment (MihaiBojin) [inline GNUmakefile:10]: an echo / echo " ==> Installing dependencies" would be nice here

**Review by MihaiBojin — APPROVED** [review]

File: GNUmakefile (hunk 6 above)
Comment (gssbzn) [reply GNUmakefile]: will do 👍

File: .drone.yml (hunk 5 above)
Comment (gssbzn) [reply .drone.yml:21]: no as it doesn't install anything, the `golangci-lint` comes with its own dependencies already build and just runs against the code

**Pull request merged**